src/gamecontrol.cpp

```diff
@@ -234,54 +270,69 @@ void GameControl::QuitGame()
-	_world.SetTileOwnerRect(9, 1,  1,  1, OWN_PARK);
-	_world.SetTileOwnerRect(2, 18, 16, 2, OWN_FOR_SALE);
-
-	std::vector<const SceneryType*> park_entrance_types = _scenery.GetAllTypes(SCC_SCENARIO);
-	if (park_entrance_types.size() < 3) {
-		_world.SetTileOwnerRect(8, 0, 4, 2, OWN_PARK); // Allow building path to map edge in north west.
+
+	if (this->next_scenario != nullptr) {
+		_scenario = this->next_scenario->scenario;
+		_scenario.wrapper = this->next_scenario;
+		_scenario.name  = _language.GetSgText(this->next_scenario->name);
+		_scenario.descr = _language.GetSgText(this->next_scenario->descr);
 	} else {
-		/* Assemble a park entrance and some paths. This assumes that the entrance parts are the first three scenario scenery items loaded. */
-		_world.AddEdgesWithoutBorderFence(Point16(9, 0), EDGE_SE);
-		for (int i = 0; i < 3; i++) {
-			SceneryInstance *item = new SceneryInstance(park_entrance_types[i]);
-			item->orientation = 0;
-			item->vox_pos.x = 8 + i;
-			item->vox_pos.y = 1;
-			item->vox_pos.z = (i == 1 ? 12 : 8);
-			_scenery.AddItem(item);
-		}
-		for (int i = 0; i < 6; i++) {
-			BuildFlatPath(XYZPoint16(9, i, 8), PAT_CONCRETE, PAS_NORMAL_PATH, false, false);
+		/* When not loading a scenario, create some default objects. \todo Clean this mess up. */
+
+		_scenario.SetDefaultScenario();
+
+		_world.SetWorldSize(WORLD_X_SIZE - 1, WORLD_Y_SIZE - 1);
+		_world.MakeFlatWorld(8);
+		_world.SetTileOwnerGlobally(OWN_NONE);
+		_world.SetTileOwnerRect(2, 2, 16, 15, OWN_PARK);
+		_world.SetTileOwnerRect(9, 1,  1,  1, OWN_PARK);
+		_world.SetTileOwnerRect(2, 18, 16, 2, OWN_FOR_SALE);
+
+		std::vector<const SceneryType*> park_entrance_types = _scenery.GetAllTypes(SCC_SCENARIO);
+		if (park_entrance_types.size() < 3) {
+			_world.SetTileOwnerRect(8, 0, 4, 2, OWN_PARK); // Allow building path to map edge in north west.
+		} else {
+			/* Assemble a park entrance and some paths. This assumes that the entrance parts are the first three scenario scenery items loaded. */
+			_world.AddEdgesWithoutBorderFence(Point16(9, 0), EDGE_SE);
+			for (int i = 0; i < 3; i++) {
+				SceneryInstance *item = new SceneryInstance(park_entrance_types[i]);
+				item->orientation = 0;
+				item->vox_pos.x = 8 + i;
+				item->vox_pos.y = 1;
+				item->vox_pos.z = (i == 1 ? 12 : 8);
+				_scenery.AddItem(item);
+			}
+			for (int i = 0; i < 6; i++) {
+				BuildFlatPath(XYZPoint16(9, i, 8), PAT_CONCRETE, PAS_NORMAL_PATH, false, false);
+			}
 		}
 	}
 
 	_inbox.Clear();
-	_finances_manager.SetScenario(_scenario);
 	_date.Initialize();
 	_weather.Initialize();
 	_game_observer.Initialize();
 }
 
-/** Initialize common game settings and view. */
-void GameControl::StartLevel()
+/**
+ * Initialize common game settings and view.
+ * @oaram editor Launch the game as a scenario editor session.
+ */
+void GameControl::StartLevel(bool editor)
 {
-	_game_mode_mgr.SetGameMode(GM_PLAY);
-	this->speed = GSP_1;
+	_game_mode_mgr.SetGameMode(editor ? GM_EDITOR : GM_PLAY);
```

Comment:
I wonder if we get other editors in the future, eg a map editor?  a pre-built ride editor?

`editor` as name is perhaps a bit too generic?

One way out is to make it an enum I guess.